### ISSUE TITLE
Password fields should have reveal "eye"

### ISSUE THREAD

**Issue by Chris-Petty** [openmsupply/open-msupply#654]
On the intialisation page and login (perhaps elsewhere) you should be able to reveal the password you are typing to make sure you'd done it correctly. Today I was using a machine with a faulty keyboard, where sometimes keystrokes don't register. I could not verify I had typed the password right without typing it elsewhere then copy pasting it into the field heh.

<details>
<summary>Long ramble stumbling it's way to a conclusion</summary>

[This article](https://css-tricks.com/the-options-for-password-revealing-inputs/) goes over the gist of using `<input type="password">...` and the classic perhaps best way of just removing that "type" (and the pitfalls of doing so even as late as last year). I reckon there should be a common stateful component for "PasswordInput" or there abouts that has an [eye icon](https://www.flaticon.com/free-icons/eye-password) on the right side of the field that reflects whether or not the password text is revealed.

**WELL** I went and looked at materialUI as that's what's in use here. Turns out they basically already have a component for this, I think it fits the bill exactly! https://mui.com/material-ui/react-text-field/#input-adornments

Also @jmbrunskill sez he has done something similar in HSH. He may want to comment. Too late I found it:
</details>

https://github.com/openmsupply/health-supply-hub/blob/1c8eed03380eefb6bf07f57ff4d21f5d3b03f76e/frontend/packages/system/src/Users/ListView/UserAccountEditModal/UserAccountEditModal.tsx#L259-L270

```ts
<BasicTextInput
  type={showPassword ? 'text' : 'password'}
  value={password || ''}
  onChange={e => setPassword(e.target.value)}
  label={t('heading.password')}
  InputProps={{
    endAdornment: visibilityInputButton,
  }}
  InputLabelProps={{
    shrink: true,
  }}
/>
```

Basically extract that into its own reusable component, use it around the place. It'll be siiiick.
